catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/SourcePollerRunner.java

```diff
@@ -96,41 +94,36 @@ public void run() {
      * Checks if the specified source is available, updating the internally maintained map of
      * sources and their status. Lock ensures only one status thread is running per source.
      *
-     * @param source
-     *            the source to check if it is available
+     * @param source the source to check if it is available
      */
     private void checkStatus(final Source source) {
 
         if (pool == null) {
             pool = Executors.newCachedThreadPool();
 
         }
-        final Runnable statusRunner = new Runnable() {
-
-            public void run() {
-
-                final CachedSource cachedSource = cachedSources.get(source);
-                if (cachedSource != null) {
-                    Lock sourceStatusThreadLock = sourceStatusThreadLocks.get(source);
-
-                    if (sourceStatusThreadLock.tryLock()) {
-                        LOGGER.debug("Acquired lock for Source [{}] with id [{}]",
+        final Runnable statusRunner = () -> {
+            final CachedSource cachedSource = cachedSources.get(getSourceKey(source));
+            if (cachedSource != null) {
+                Lock sourceStatusThreadLock = sourceStatusThreadLocks.get(cachedSource);
+
+                if (sourceStatusThreadLock.tryLock()) {
+                    LOGGER.debug("Acquired lock for Source [{}] with id [{}]",
```

Comment:
This debug statement could still throw an exception if the underlying source is gone. Should probably use `getSourceKey(source).getId()`. Might also be best to move it inside the `try` block so it can't prevent the lock from being released.